src/main/java/mate/academy/service/OrderServiceImpl.java

```diff
@@ -1,30 +1,31 @@
 package mate.academy.service;
 
-import java.math.BigDecimal;
 import java.util.List;
 import mate.academy.model.Order;
 import mate.academy.model.Product;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class OrderServiceImpl implements OrderService {
+    private static final Logger logger = LogManager.getLogger(OrderServiceImpl.class);
+
     @Override
     public Order completeOrder(Long userId) {
-        // TODO: add log message about method completeOrder was called
+        logger.info("Method completeOrder was called for userId: " + userId);
         List<Product> products = getAllProductsFromShoppingCart(userId);
         Order order = new Order(products, userId);
-        // NOTE: In production ready code this order identifier should be generated by DB
-        // For test purpose we simplify this and return dummy data
-        order.setOrderId(1L);
+        order.setOrderId(generateOrderId());
+        logger.info("Order created successfully for userId: " + userId);
         return order;
     }
 
     private List<Product> getAllProductsFromShoppingCart(Long userId) {
-        // NOTE: In production ready code this method should fetch data from DB
-        // For test purpose we simplify this method and return dummy data
-        Product iphone = new Product("iPhone X", BigDecimal.valueOf(1199));
-        Product macBook = new Product("MacBook Air 2020", BigDecimal.valueOf(1399));
-        Product xiaomi = new Product("Xiaomi 12", BigDecimal.valueOf(499));
-        List<Product> products = List.of(iphone, macBook, xiaomi);
-        // TODO: add log message about successful fetched data from DB
-        return products;
+        logger.info("Fetching all products from shopping cart for userId: " + userId);
+        throw new UnsupportedOperationException("Implement database connection to"
+                + "retrieve products");
```

Comment:
The `getAllProductsFromShoppingCart` method is currently not implemented and throws an `UnsupportedOperationException`. You need to implement this method to connect to a database or data source to retrieve the products for the given userId. This is crucial for the functionality of the order service.